boudicca.events/eventcollectors/src/main/kotlin/events/boudicca/eventcollector/collectors/OKHVoecklabruckCollector.kt

```diff
@@ -0,0 +1,68 @@
+package events.boudicca.eventcollector.collectors
+
+import base.boudicca.SemanticKeys
+import base.boudicca.api.eventcollector.Fetcher
+import base.boudicca.api.eventcollector.TwoStepEventCollector
+import base.boudicca.model.Event
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import java.time.*
+import java.time.format.DateTimeFormatter
+
+class OKHVoecklabruckCollector : TwoStepEventCollector<Pair<String, String>>("okhvoecklabruck") {
+    private val fetcher = Fetcher()
+    private val baseUrl = "https://www.okh.or.at/"
+
+    override fun getAllUnparsedEvents(): List<Pair<String, String>> {
+        val events = mutableListOf<Pair<String, String>>()
+        val document = Jsoup.parse(fetcher.fetchUrl(baseUrl + "programm"))
+        document.select("div.event_box").forEach {
+            events.add(Pair(
+                it.select("a.event-inner").attr("href"),
+                it.select("div.event-art").text()
+            ))
+        }
+        return events
+    }
+
+    override fun parseEvent(event: Pair<String, String>): Event {
+        val (eventUrl, eventType) = event
+        val url = baseUrl + eventUrl
+        val eventSite = Jsoup.parse(fetcher.fetchUrl(url))
+
+        val name = eventSite.select("h1").text()
+        val startDate = parseDate(eventSite)
+
+        val data = mutableMapOf<String, String>()
+        data[SemanticKeys.URL] = url
+        data[SemanticKeys.SOURCES] = data[SemanticKeys.URL]!!
+
+        if(eventType.isNotBlank()){
+            data[SemanticKeys.TYPE] = eventType
```

Comment:
fyi: you do not need to add those isNotBlank() checks anymore if the only reason is to avoid an empty field. i made the detection smarter :)